avocado/core/runner.py

```diff
@@ -316,7 +390,12 @@ def sigterm_handler(signum, frame):     # pylint: disable=W0613
         # STDIN fd (0), with the same fd previously set by
         # `multiprocessing.Process()`
         os.dup2(sys.stdin.fileno(), 0)
-
+        # Replace the 1 and 2 with our pipe, which then forwards the output
+        # to the logger redirector, which produces raw messages in the "stdout"
+        # file as well as "[stdout] " prefixed ones in avocado.test logger.
```

Comment:
To keep things simpler, at the test process level, how about just redirecting the file descriptors to `stdout` and `stderr` at `$test-result-dir/` ? This would require the test runner, who is aware of the test result output location, to read directly from those files as they're written by the test process.  Maybe this is a more rustic approach, but I think it'd simplify the current implementation greatly.

Reply:
You actually have to pipe all records (not just messages) to the runner as `--show` allows you to show/record any kind of logger. So we have multiple options here:

1. tweak test's logging to always emit all records to runner and postprocess them from there (+- impossible)
2. tweak test's logging to emit all avocado-configured logging streams to runner and postprocess them from there (basically what I called the proper solution here)
3. tweak test's logging to produce raw messages while replacing the fd1 and fd2 (this PR)
4. disable direct stdout output in test process and read/print the output of it's output files (your suggestion)

(1) is impossible because user can re-initialize logging from within the test
(2) allows to keep all the current functionality as we know in advance which streams are to be stored/displayed, but it passes all messages over to runner, which is not really efficient.
(3) the same problem as (1), user can spoil the output, but in such case it'd only result in those messages ending in test's stdout/stderr rather than in the expected logging stream
(4) works, but we'd have to parse which logging stream the message belonged to and forward it to the expected location (display, record, ignore).

So the (4) is simplest and people can probably live with it., the (2) is IMO the right way and everything else is just a compromise.